### PR TITLE
Add UDF support for BigQuery

### PR DESCRIPTION
**Pull request by ukarlsson**

### CODE REVIEW

File: luigi/contrib/bigquery.py [hunk 1]
```diff
@@ -594,7 +600,8 @@ def run(self):
                     'allowLargeResults': True,
                     'createDisposition': self.create_disposition,
                     'writeDisposition': self.write_disposition,
-                    'flattenResults': self.flatten_results
+                    'flattenResults': self.flatten_results,
+                    'userDefinedFunctionResources': [{"resourceUri": v} for v in self.udf_resource_uris]
```
Comment (Tarrasch) [inline luigi/contrib/bigquery.py]: Add a trailing comma please

File: luigi/contrib/bigquery.py [hunk 2]
```diff
@@ -567,6 +567,12 @@ def query_mode(self):
         """The query mode. See :py:class:`QueryMode`."""
         return QueryMode.INTERACTIVE
 
+    @property
+    def udf_resource_uris(self):
+        """List of code resource to load from a Google Cloud Storage URI (gs://bucket/path).
```
Comment (Tarrasch) [inline luigi/contrib/bigquery.py]: s/List/Iterable?

Comment (ukarlsson) [inline luigi/contrib/bigquery.py]: Fixed.

File: luigi/contrib/bigquery.py (hunk 1 above)
Comment (ukarlsson) [inline luigi/contrib/bigquery.py]: Fixed.

**Comment by ukarlsson** [comment]
I have added tests. Yes, we use this in production

**Comment by ukarlsson** [comment]
@Tarrasch OK thanks have fixed the whitespacing and rebased :)

File: luigi/contrib/bigquery.py (hunk 1 above)
Comment (Tarrasch) [inline luigi/contrib/bigquery.py]: what happened to the trailing comma you added??

**Comment by Tarrasch** [comment]
I merge as this have had many minor iterations already. But I see that @ukarlsson actually never filled out the template in the PR. While it's indeed optional, I think you could jot down some words about why your company needed this patch. What can be done now that you couldn't do before?

Thanks for the patch! :)

**Pull request merged**